modules/caddyhttp/reverseproxy/streaming.go

```diff
@@ -459,20 +459,19 @@ func (m *maxLatencyWriter) Write(p []byte) (n int, err error) {
 	defer m.mu.Unlock()
 	n, err = m.dst.Write(p)
 	if m.latency < 0 {
-		//nolint:errcheck
-		m.flush()
-		return
+		err = m.flush()
```

Comment:
This is shadowing the `err` from the `m.dst.Write` above and thus might change the logic.

It might be worth considering returning an `errors.Join` of both errors, maybe?